src/components/measureGroups/SupplementalElements.tsx

```diff
@@ -1,7 +1,381 @@
+        measure.supplementalData.map((supData) => {
+          return supData.description;
+        })
+      );
+    }
+  }, [measure]);
+
+  useEffect(() => {
+    if (measure?.supplementalData) {
+      setSelectedSupplementalData(
+        measure.supplementalData.map((supData) => {
+          return supData;
+        })
+      );
+    }
+  }, [measure]);
+
+  const formik = useFormik({
+    enableReinitialize: true,
+    initialValues: {
+      selectedSupplementalDefinitions: selectedDefinitions || [],
+      supplementalData: measure?.supplementalData || [],
+      selectedDescriptions: selectedDescriptions || [],
+    },
+    onSubmit: (values) => {
+      submitForm(values);
+    },
+  });
+  const { resetForm } = formik;
+
+  const submitForm = (values) => {
+    const submitMeasure = {
+      ...measure,
+      supplementalData: formik.values.supplementalData,
+    };
+
+    measureServiceApi
+      .updateMeasure(submitMeasure)
+      .then(() => {
+        handleToast(
+          "success",
+          `Supplement Data Element Information Saved Successfully`,
+          true
+        );
+        updateMeasure(submitMeasure);
+      })
+      .catch(() => {
+        const message = `Error updating measure "${measure.measureName}"`;
+        setErrorMessage(message);
+
+        handleToast(
+          "danger",
+          `Error updating measure "${measure.measureName}"`,
+          true
+        );
+      });
+  };
+
+  useEffect(() => {
+    const subscription = measureStore.subscribe(setMeasure);
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  // Route handle store will give warning message
+  // if form is dirty and user ties to navigate across SPAs
+  const { updateRouteHandlerState } = routeHandlerStore;
+  const [discardDialogOpen, setDiscardDialogOpen] = useState(false);
+
+  useEffect(() => {
+    updateRouteHandlerState({
+      canTravel: !formik.dirty,
+      pendingRoute: "",
+    });
+  }, [formik.dirty]);
+
+  const handleDefinitionChange = (selectedValues: string[]) => {
+    const oldSelectedSupDataList = cloneDeep(selectedSupplementalData);
+
+    const newList: SupplementalData[] = [];
+    selectedValues.forEach((selectedDefinition) => {
+      newList.push({
+        definition: selectedDefinition,
+        description: getDescriptionByDefinition(
+          selectedDefinition,
+          oldSelectedSupDataList
+        ),
+      });
+    });
+
+    setSelectedSupplementalData(newList);
+    formik.setFieldValue("supplementalData", newList);
+  };
+  const getDescriptionByDefinition = (
+    definition: string,
+    oldSelectedSupDataList: SupplementalData[]
+  ): string => {
+    let description = "";
+    oldSelectedSupDataList.forEach((selected) => {
+      if (definition === selected.definition) {
+        description = selected.description;
+      }
+    });
+    return description;
+  };
+
+  const handleDesciptionChange = (definition, value, index) => {
```

Comment:
Not a biggie, but typo `handleDescriptionChange`

Reply:
@RohitKandimalla Done, thanks so much Rohit!